design-proposals/orch-cli.md

```diff
@@ -0,0 +1,229 @@
+# Design Proposal: Orchestrator Command Line Interface
+
+Author(s): Scott Baker
+
+Last updated: 2025-04-29
+
+## Abstract
+
+This ADR describes the design of a command line interface (CLI) for the
+Orchestrator. The CLI is a cross-component tool that may be used from a Linux, Windows, or
+Mac environment, such as from the administrator's laptop. The CLI may also may be leveraged
+as a building block in other environments such as CI/CD or testing.
+
+The CLI described here uses the `Catalog CLI` as a strating point and inspiration. However, this
+ADR assumes that syntax and semantics of the CLI may diverge from the existing `Catalog CLI` and
+this ADR is not constrained by that existing implementation. As such, examples given in this
+ADR may not necessarily work with the existing catalog CLI.
+
+## Proposal
+
+The CLI implements a set of verbs, a set of nouns, and a set of common options that may
+be used to interact with the orchestrator.
+
+The CLI follows a `verb - noun - subject` pattern. For example,
+
+- `cli get application nginx`. The verb is `get`. The noun is `application` and the subject
+  is `nginx`. This particular CLI command would return the application called nginx from the
+  Orchestrator and print them to the console in a human-readable format.
+
+### Syntax
+
+This section describes the syntax of the CLI.
+
+#### Verbs
+
+Note that some verbs may have synonyms. For example, `set` and `update` are synonyms for the same
+verb.
+
+##### Session Management
+
+- `login <user-id>` ... log in to the Orchestrator. The CLI shall prompt for a password and retrieve
+  a refresh token. The refresh token is cached locally until it is expired or a `logout` verb is used.
+
+- `logout` ... log out of the Orchestrator. Any active login state, including the refresh token, is
+  discarded.
+
+##### CRUD Operations
+
+- `list` ... lists objects on the Orchestrator. The `list` verb produces a list of all objects
+  on the Orchestrator, with an optional filter. List should return tabular summary-level information
+  about the objects.
+
+- `get` / `describe` ... retrieves objects from the Orchestrator. `get` is for retrieving single objects whereas
+  `list` is for retrieving multiple objects. `get` should retrieve more verbose information about the
+  object than `list`.
+
+- `create` ... creates an object on the Orchestrator. Additional noun-specific parameters are used to
+  describe the properties of the object to be created.
+
+- `delete` ... deletes an object from the Orchestrator.
+
+- `set` / `update` ... modifies one or more values in an existing object on the Orchestrator.
+
+- `apply` ... create or modify an object from a yaml or json specification, from stdin or from a local
+  file.
+
+##### Configuration and utility
+
+- `config` ... configures the CLI. For example, `cli config set endpoint https://my-orchestrator.com/`
```

Comment:
`config` could be handled with `set-config` and `get-config` verbs.

It is about configuring the CLI and not sent to the backend. Settings are stored in the config file as mentioned below.

Reply:
Updated to use `get config` and `update config` verb/noun pairs.